pkg/provider/amazon.go

```diff
@@ -0,0 +1,254 @@
+package provider
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/alibabacloud-go/tea/tea"
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/credentials"
+	"github.com/aws/aws-sdk-go-v2/service/route53"
+	"github.com/aws/aws-sdk-go-v2/service/route53/types"
+	"github.com/aws/aws-sdk-go-v2/service/route53domains"
+	domainTypes "github.com/aws/aws-sdk-go-v2/service/route53domains/types"
+	"github.com/eryajf/cloud_dns_exporter/public"
+	"github.com/golang-module/carbon/v2"
+	"golang.org/x/net/context"
+	"strings"
+	"sync"
+	"time"
+)
+
+type AmazonDNS struct {
+	account public.Account
+	client  *route53.Client
+}
+
+const region = "us-east-1"
+
+func NewAwsDnsClient(secretID string, secretKey string) *route53.Client {
+	return route53.New(route53.Options{
+		Credentials:      aws.NewCredentialsCache(credentials.NewStaticCredentialsProvider(secretID, secretKey, "")),
+		Region:           region,
+		RetryMaxAttempts: 3,
+	})
+}
+
+func NewAwsDns(account public.Account) *AmazonDNS {
+	client := NewAwsDnsClient(account.SecretID, account.SecretKey)
+	return &AmazonDNS{
+		account: account,
+		client:  client,
+	}
+}
+
+func NewAwsDomainClient(secretID string, secretKey string) *route53domains.Client {
+	return route53domains.New(route53domains.Options{
+		Credentials: aws.NewCredentialsCache(credentials.NewStaticCredentialsProvider(secretID, secretKey, "")),
+		Region:      region,
+	})
+}
+
+func (a *AmazonDNS) ListDomains() ([]Domain, error) {
+	ad := NewAwsDns(public.Account{
+		CloudProvider: a.account.CloudProvider,
+		CloudName:     a.account.CloudName,
+		SecretID:      a.account.SecretID,
+		SecretKey:     a.account.SecretKey,
+	})
+	a.client = ad.client
+	var dataObj []Domain
+	domains, err := a.getDomainList()
+	if err != nil {
+		return nil, err
+	}
+	domainNames, err := a.getDomainNameList()
+	for _, domain := range domains {
+		domainName := strings.TrimSuffix(*domain.Name, ".")
```

Comment:
```suggestion
		domainName := strings.TrimSuffix(*domain.Name, ".")
```

为了避免出现panic，这里去Name的时候最好也使用tea.StringValue()来取